base/src/cache.rs

```diff
@@ -201,22 +238,33 @@ impl<T: Backend> CacheUpdate<T> for Event {
             Event::ShardReconnecting(_) => future::ok(()).boxed(),
             Event::ShardResuming(_) => future::ok(()).boxed(),
             Event::TypingStart(_) => future::ok(()).boxed(),
-            Event::UnavailableGuild(_) => todo!(),
-            _ => todo!(),
+            Event::UnavailableGuild(_) => future::ok(()).boxed(),
+            Event::UserUpdate(event) => event.process(cache),
+            Event::VoiceServerUpdate(_) => future::ok(()).boxed(),
+            Event::VoiceStateUpdate(event) => event.process(cache),
+            Event::WebhooksUpdate(_) => future::ok(()).boxed(),
         }
     }
 }
 
-impl<T: Backend> CacheUpdate<T> for ChannelCreate {
+impl<T: Backend + Send + Sync> CacheUpdate<T> for ChannelCreate {
     fn process<'a>(
         &'a self,
         cache: &'a Cache<T>,
     ) -> Pin<Box<dyn Future<Output = Result<(), T::Error>> + Send + 'a>> {
         match &self.0 {
             Channel::Group(group) => {
+                let futures = FuturesUnordered::new();
+
+                for user in &group.recipients {
+                    let entity = UserEntity::from(user.clone());
+                    futures.push(cache.users.upsert(entity));
+                }
```

Comment:
This can be
```rust
FuturesUnordered::from_iter(group.recipients.clone().into_iter().map(|user|  {
    cache.users.upsert(UserEntity::from(user))
})
```
which is arguably simpler